src/components/views/elements/ImageView.tsx

```diff
@@ -160,41 +160,82 @@ export default class ImageView extends React.Component<IProps, IState> {
         });
     };
 
-    private zoom(delta: number) {
-        const newZoom = this.state.zoom + delta;
+    private zoomDelta(delta: number, zoomX?: number, zoomY?: number) {
+        this.zoom(this.state.zoom + delta, zoomX, zoomY);
+    }
+
+    private zoom(zoomLevel: number, zoomX?: number, zoomY?: number) {
+        const oldZoom = this.state.zoom;
+        const newZoom = Math.min(zoomLevel, this.state.maxZoom);
 
         if (newZoom <= this.state.minZoom) {
+            // Zoom out fully
             this.setState({
                 zoom: this.state.minZoom,
                 translationX: 0,
                 translationY: 0,
             });
-            return;
-        }
-        if (newZoom >= this.state.maxZoom) {
-            this.setState({ zoom: this.state.maxZoom });
-            return;
-        }
+        } else if (zoomX === undefined && zoomY === undefined) {
+            // Zoom relative to the center of the view
+            this.setState({
+                zoom: newZoom,
+                translationX: this.state.translationX * newZoom / oldZoom,
+                translationY: this.state.translationY * newZoom / oldZoom,
```

Comment:
I don't think I understand why are we doing this, maybe I should just go to sleep :sweat_smile:

Reply:
If you leave `translationX`/`translationY` untouched, then it will appear to zoom in on the center of the image, which can cause the part of the image you were looking at to slide away if it's far away from the center. In order to actually keep the center of the *current view* fixed in place, you have to scale the translation to match the new zoom level. That's what this is doing.

Reply:
Thanks for explaining - makes sense!